### ISSUE TITLE
Update jvm version to 22

### ISSUE THREAD

**Issue by corneliusroemer** [loculus-project/loculus#3148]
Gradle 8.8 and Kotlin 2.0.20 support Java 22 so we can probably upgrade our Java version now in backend (this is something dependabot doesn't do). If someone's bored.

**Comment by fengelniederhammer** [comment]
I would prefer staying on Java 21, because it's LTS (less maintenance). We don't need new Java features, we just depend on the JVM.

We can then go to Java 25, once it's released (the next LTS).

**Comment by chaoran-chen** [comment]
I agree with Fabian and don't see a good reason to update, so I'll close this. @corneliusroemer, feel free to reopen if you think that we should update now.
